### PR TITLE
Enable PHPStan Extensions 

### PR DESCRIPTION
**Pull request by mabumusa1**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

This PR introduce PHPStan extensions that allows a better scanning to the code base and stop any technical debt in future PRs, these extensions allows PHPStan to understand `symfony`, `phpunit`, `deprecation`, and `php parser`. without those extensions we had to ignore some errors. 

PHPStan baseline will be changed in this PR as some errors will be removed and others will be added, I will start an issue with the log of phpstan run, so we can remove those issues as we go 

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Only code review is required

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -167,11 +167,15 @@ jobs:
     - name: Install dependencies
       run: |
         composer validate
-        composer install --prefer-dist --no-progress --no-suggest
```
Comment (mollux) [inline .github/workflows/tests.yml]: @mabumusa1 is there a reason `--prefer-dist` is not possible anymore? as I see no composer changes that require this

File: tests/console-application.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    }
+}
+$output = (false !== $input->getParameterOption(['--no-interaction', '-n'])) ? new ConsoleDatetimeOutput() : null;
+
+$kernel      = new AppKernel($_SERVER['APP_ENV'], (bool) $_SERVER['APP_DEBUG']);
+$application = new Application($kernel);
+$application->setName('Mautic');
+$application->setVersion($kernel->getVersion().' - app/'.$kernel->getEnvironment().($kernel->isDebug() ? '/debug' : ''));
+
+return $application;
```
Comment (escopecz) [inline tests/console-application.php]: I don't like that this whole file is duplicated except the last line. But TBH, I'm having a hard time suggesting a better solution. Anyone?

File: .github/workflows/tests.yml (hunk 1 above)
Comment (mabumusa1) [reply .github/workflows/tests.yml]: @mollux good catch, I changed this line while testing, and reverted back on commit c33aa03

**Review by escopecz — DISMISSED** [review]
We can give this a try, but I'm afraid it it might make development for Mautic very painful. We have some deprecated code that have no alternatives. So developers are basically forced to use deprecated code. But perhaps after a few months of pain it will force us to either un-deprecate some code or make an alternative to it so developers don't have to use the deprecated code.

We can always comment out some rules if they become a blocker to something. Let's try this 👍

File: tests/console-application.php (hunk 2 above)
Comment (fedys) [reply tests/console-application.php]: @escopecz I addressed the duplicity in the following [commits](https://github.com/mautic/mautic/pull/11970/files/c33aa03593d2627c99eb6703baed094d4b59a880..3357a92f06a1980a781de63c70cb756f003235a8).

File: phpstan.neon [hunk 3]
```diff
@@ -34,4 +34,7 @@ parameters:
 	doctrine:
 		objectManagerLoader: tests/object-manager.php
 		allowNullablePropertyForRequiredField: true
+	symfony:
+		containerXmlPath: var/cache/dev/AppKernelDevDebugContainer.xml
+		consoleApplicationLoader: app/console-application.php
```
Comment (fedys) [inline phpstan.neon:39]: @mabumusa1 I tried changing the value of `symfony.consoleApplicationLoader` to a non-existent path and ran `composer phpstan` in my local env. Interestingly no warning or error popped up. It makes me thinking whether this configuration is needed?

Comment (escopecz) [inline phpstan.neon:39]: @mabumusa1 is this even necessary? We had a call with @fedys and it seems that PHPSTAN runs successfully even without this line.

Comment (mabumusa1) [reply phpstan.neon:39]: @escopecz the first line is needed according to the [docs here](https://github.com/phpstan/phpstan-symfony#configuration)

For the 2nd line, it is need to analyze the commands as [documented here  ](https://github.com/phpstan/phpstan-symfony#analysis-of-symfony-console-commands)

So I think they are needed

**Review by escopecz — APPROVED** [review]
👍

**Review by fedys — COMMENTED** [review]
👍

File: phpstan.neon (hunk 3 above)
Comment (escopecz) [reply phpstan.neon:39]: Already asked in https://github.com/mautic/mautic/pull/11970#discussion_r1224193146.

**Pull request merged**